### ISSUE TITLE
Sanity-check importance of regression and gradient descent for optimizing the stitching layer

### ISSUE THREAD

**Issue by wrongu** [pkawale/StitchNet#6]
Initial results from M1 report indicate that test loss does not change during gradient descent *provided that the stitching layer was initialized using linear regression*. There are two possibilities:

1. linear regression is already so close to the optimal solution that there is nothing left to learn by gradient descent
2. there's some other issue in gradient descent causing it not to learn

Option 2 seems unlikely given that training loss does decrease. But we should sanity-check and confirm. Suggested check: compare loss curves with vs without regression initialization. Expected result: naive (random) initialization of stitching layer converges over many training iterations to match the performance of the regression approach.

check this for multiple models x layers.
